pkg/model/iam/iam_builder.go

```diff
@@ -736,21 +736,33 @@ func addCalicoSrcDstCheckPermissions(p *Policy) {
 func AddAWSLoadbalancerControllerPermissions(p *Policy, resource stringorslice.StringOrSlice, clusterName string) {
 	addMasterEC2Policies(p, resource, false, clusterName)
 	addMasterELBPolicies(p, resource, false)
-	p.Statement = append(p.Statement, &Statement{
-		Effect: StatementEffectAllow,
-		Action: stringorslice.Of(
-			"ec2:AuthorizeSecurityGroupIngress", // aws.go
-			"ec2:DeleteSecurityGroup",           // aws.go
-			"ec2:RevokeSecurityGroupIngress",    // aws.go
-		),
-		Resource: resource,
-		Condition: Condition{
-			"StringEquals": map[string]string{
-				"ec2:ResourceTag/elbv2.k8s.aws/cluster": clusterName,
-			},
+	p.Statement = append(p.Statement,
+		&Statement{
+			Effect: StatementEffectAllow,
+			Action: stringorslice.Of(
+				"ec2:DescribeAvailabilityZones",
+			),
+			Resource: stringorslice.Slice([]string{"*"}),
 		},
-	},
+		&Statement{
+			Effect: StatementEffectAllow,
+			Action: stringorslice.Of(
+				"ec2:AuthorizeSecurityGroupIngress", // aws.go
+				"ec2:DeleteSecurityGroup",           // aws.go
+				"ec2:RevokeSecurityGroupIngress",    // aws.go
 
+				"elasticloadbalancing:ModifyTargetGroupAttributes",
+				"elasticloadbalancing:ModifyRule",
+				"elasticloadbalancing:DeleteRule",
+				"elasticloadbalancing:RemoveTags",
```

Comment:
`RemoveTags` is being added in both statements. Can we consolidate it to just one statement?

Reply:
Hm. Is the last statement redundant? Technically they should be for different types of resources (ELBs vs listeners)

Reply:
Think this should work just as well.